### PR TITLE
Cloudsiem Quickstart Version 2.0 Updates

### PR DESCRIPTION
**Pull request by himanshu219**
1. Upgrade Python and NodeJS runtime
2. Fixes for CFN lint, CFN Nag and bridgecrew
3. Added versioning 
4. Upgraded SAM applications
5. Upgraded dependencies in lambda helper function for CVE fixes
6. Bug fixes for Cloudtrail collection for bucket path expression
7. Bug fixes for Network Firewall Collection to enable AWS Config and AWS RAM
8. Updated Centralized collection for Security Hub using single configuration policy.

### CODE REVIEW

File: .taskcat.yml
Comment (gcasilva) [inline .taskcat.yml:2]: please remove "/2" from project name in taskcat

File: templates/apps/sumo-aws-apps/cloudtrail-org/cloudtrail-main.yaml
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/cloudtrail-org/cloudtrail-main.yaml:339]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/cloudtrail-org/cloudtrail-main.yaml
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/cloudtrail-org/cloudtrail-main.yaml:364]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml:493]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml:575]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml:618]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml:655]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml:779]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-configuration.yaml
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-configuration.yaml:319]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-configuration.yaml
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-configuration.yaml:338]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-configuration.yaml
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-configuration.yaml:383]: remove ${QSVersion}

**Review by gcasilva — COMMENTED** [review]
Please remove ${QSVersion} from Template URLs, check in the comments.
Also remove /2 from project name in taskcat.yml, check in comments.

File: .taskcat.yml
Comment (gcasilva) [reply .taskcat.yml:2]: Also uncomment and change lines 19-22 so we can use in the automated tests by referencing SSM Parameter Store for these variables in our test AWS Account:

Section1SumoLogicDeployment: "us2"
Section1SumoLogicAccessID: $[taskcat_ssm_/sumologic/Section1SumoLogicAccessID]
Section1SumoLogicAccessKey: $[taskcat_ssm_/sumologic/Section1SumoLogicAccessKey]
Section1SumoLogicOrganizationId: $[taskcat_ssm_/sumologic/Section1SumoLogicOrganizationId]

File: .taskcat.yml
Comment (gcasilva) [inline .taskcat.yml:17]: change to us-east-1 for our automated tests.

File: .taskcat.yml
Comment (gcasilva) [inline .taskcat.yml:24]: change to us-east-1 for our automated tests.

File: .taskcat.yml
Comment (gcasilva) [inline .taskcat.yml:37]: change to us-east-1 for our automated tests.

File: .taskcat.yml
Comment (gcasilva) [inline .taskcat.yml:51]: change to us-east-1 for our automated tests.

File: .taskcat.yml
Comment (gcasilva) [inline .taskcat.yml:63]: change to us-east-1 for our automated tests.

**Comment by gcasilva** [comment]
@himanshu219 please change all regions in .taskcat.yml to us-east-1 for our automated tests to execute properly in our test account.

**Comment by gcasilva** [comment]
@himanshu219 can you please confirm if there's any additional commits to be added to this PR before we do final review/tests?

**Comment by himanshu219** [comment]
> @himanshu219 can you please confirm if there's any additional commits to be added to this PR before we do final review/tests?

There might be some commits related to documentation.

**Comment by gcasilva** [comment]
@himanshu219 please update me here when all commits have been sent so I can do final review and test. Thanks

**Comment by himanshu219** [comment]
@gcasilva  yes all the commits are sent for both the quickstart PRs

File: templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml:549]: @himanshu219 
I'm getting an error on this resource, it seems like the policy for the IAM Role for this resource needs to be updated to add UpdateOrganizationConfiguration API from Security Hub, please check and fix:
"Received response status [FAILED] from custom resource. Message returned: An error occurred (AccessDeniedException) when calling the UpdateOrganizationConfiguration operation: You don't have permissions to access this resource."

**Review by gcasilva — COMMENTED** [review]
@himanshu219 I'm having an error on SecurityHubStack stack please check my comment to review/fix.

File: templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml
Comment (langmtt) [reply templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml:549]: @gcasilva Please share the values of parameters: Section4SecurityHubAggregationRegion, Section4SecurityHubLinkedRegions

Comment (gcasilva) [reply templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml:549]: @langmtt Below are the value of the parameters. FYI these regions are also correctly enabled as governed regions in Control Tower.
Section4SecurityHubAggregationRegion: us-east-1
Section4SecurityHubLinkedRegions: us-east-2, us-west-1

Comment (gcasilva) [reply templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml:549]: @himanshu219 Tested successfully the deployment with the fix sent. Can you please confirm if there's any additional documentation changes you'll add to this PR before we close it and merge it to main?
Thanks

Comment (himanshu219) [reply templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml:549]: @gcasilva  nops, there is no doc changes required.

**Review by gcasilva — DISMISSED** [review]

File: .taskcat.yml
Comment (gcasilva) [inline .taskcat.yml:23]: @langmtt please change this to the below so we can execute the automated tests to merge:
Section1SecurityToolingAccountId: $[taskcat_ssm_/sumologic/Section1SecurityToolingAccountId]

File: .taskcat.yml
Comment (gcasilva) [inline .taskcat.yml:25]: @langmtt please change this to the below so we can execute the automated tests to merge:
Section1LogArchivingAccountId: $[taskcat_ssm_/sumologic/Section1LogArchivingAccountId]

File: .taskcat.yml
Comment (gcasilva) [inline .taskcat.yml:26]: @langmtt please change this to the below so we can execute the automated tests to merge:
Section1OrganizationRootID: $[taskcat_ssm_/sumologic/Section1OrganizationRootID]

**Comment by himanshu219** [comment]
@gcasilva please let us know if there are any action items on us for these failing e2e tests.

**Comment by gcasilva** [comment]
> @gcasilva please let us know if there are any action items on us for these failing e2e tests.
@himanshu219: please fix the below in taskcat.yml 
Section6DeliveryStreamName is failing in the automated test due to length validation in the template (16 character max). Please change the value to 'sumologic' so it matches the default in the template

**Review by gcasilva — APPROVED** [review]

**Review by langmtt — APPROVED** [review]

File: .taskcat.yml
Comment (himanshu219) [reply .taskcat.yml]: @gcasilva  is it still failing due to length `sumo-kinesis` is 12 chars long ?

Comment (himanshu219) [reply .taskcat.yml]: @gcasilva  any update on above ? we have verified by testing with taskcat also and locally as well. It seems to be passing for us. Could you please let us know why the e2e-tests are failing, is it with same error or some different error?

**Review by gcasilva — CHANGES_REQUESTED** [review]
@himanshu219 @langmatt please check review for changes needed due to e2e failing

**Comment by gcasilva** [comment]
@himanshu219 I'll merge this manually tomorrow with the AWS Quickstarts team and I can work on fixing whatever is needed for the e2e deployment later since the template is working appropriately from customer perspective. Thanks

**Comment by langmatt** [comment]
Am I supposed to be on this?

**Review by gcasilva — APPROVED** [review]

**Pull request merged**
